plugins/plugin-typescript/plugin.js

```diff
@@ -0,0 +1,32 @@
+const execa = require('execa');
+const npmRunPath = require('npm-run-path');
+const cwd = process.cwd();
+
+function typescriptPlugin() {
+  return {
+    name: '@snowpack/plugin-typescript',
+    async run({isDev, log}) {
+      const workerPromise = execa.command(`tsc --noEmit ${isDev ? '--watch' : ''}`, {
```

Comment:
Would you like to use the programmatic API instead of parsing the output?

Reply:
Not opposed to the JS API, but in the past I know that the project has always treated the JS API as secondary to the CLI itself. 

On top of that, the CLI gives us the formatted output for free. Since the dev console just wants formatted output and a simple pass/fail, the tsc CLI is enough for our use-case and lets the tsc CLI handle all of the complexity of config loading, file watching, etc.

Reply:
So tsc is only used for type checking, not file emitting?

Reply:
👍 That's correct

Reply:
reading over the README, we could probably make that difference even more clear

Reply:
Oh I noticed the --noEmit. IMO it's a bit wasty to use multiple tool to parse the source code. In my project I mainly don't use babel and do everything only with TypeScript compiler.

Reply:
If you wanted to build a typescript build plugin, that would be exciting to see!

Snowpack doesn't run the TypeScript compiler as a part of the default build step, we use esbuild which can parse TS code successfully. So this plugin only exists to do that typechecking for you.